velox/exec/Spill.cpp

```diff
@@ -449,33 +452,74 @@ SpillStats& SpillStats::operator+=(const SpillStats& other) {
   return *this;
 }
 
+SpillStats SpillStats::operator-(const SpillStats& other) const {
+  SpillStats result;
+  result.spillRuns = spillRuns - other.spillRuns;
+  result.spilledBytes = spilledBytes - other.spilledBytes;
+  result.spilledRows = spilledRows - other.spilledRows;
+  result.spilledPartitions = spilledPartitions - other.spilledPartitions;
+  result.spilledFiles = spilledFiles - other.spilledFiles;
+  result.spillFillTimeUs = spillFillTimeUs - other.spillFillTimeUs;
+  result.spillSortTimeUs = spillSortTimeUs - other.spillSortTimeUs;
+  result.spillSerializationTimeUs =
+      spillSerializationTimeUs - other.spillSerializationTimeUs;
+  result.spillDiskWrites = spillDiskWrites - other.spillDiskWrites;
+  result.spillFlushTimeUs = spillFlushTimeUs - other.spillFlushTimeUs;
+  result.spillWriteTimeUs = spillWriteTimeUs - other.spillWriteTimeUs;
+  return result;
+}
+
 bool SpillStats::operator<(const SpillStats& other) const {
-  return std::tie(
-             spilledBytes,
-             spilledRows,
-             spilledPartitions,
-             spilledFiles,
-             spillFillTimeUs,
-             spillSortTimeUs,
-             spillSerializationTimeUs,
-             spillDiskWrites,
-             spillFlushTimeUs,
-             spillWriteTimeUs) <
-      std::tie(
-             other.spilledBytes,
-             other.spilledRows,
-             other.spilledPartitions,
-             other.spilledFiles,
-             other.spillFillTimeUs,
-             other.spillSortTimeUs,
-             other.spillSerializationTimeUs,
-             other.spillDiskWrites,
-             other.spillFlushTimeUs,
-             other.spillWriteTimeUs);
+  uint32_t eqCount{0};
+  uint32_t gtCount{0};
+  uint32_t ltCount{0};
+#define UPDATE_COUNTER(counter)           \
+  do {                                    \
+    if (counter < other.counter) {        \
+      ++ltCount;                          \
+    } else if (counter > other.counter) { \
+      ++gtCount;                          \
+    } else {                              \
+      ++eqCount;                          \
+    }                                     \
+  } while (0);
+
+  UPDATE_COUNTER(spillRuns);
+  UPDATE_COUNTER(spilledBytes);
+  UPDATE_COUNTER(spilledRows);
+  UPDATE_COUNTER(spilledPartitions);
+  UPDATE_COUNTER(spilledFiles);
+  UPDATE_COUNTER(spillFillTimeUs);
+  UPDATE_COUNTER(spillSortTimeUs);
+  UPDATE_COUNTER(spillSerializationTimeUs);
+  UPDATE_COUNTER(spillDiskWrites);
+  UPDATE_COUNTER(spillFlushTimeUs);
+  UPDATE_COUNTER(spillWriteTimeUs);
+#undef UPDATE_COUNTER
+  VELOX_CHECK_LE(
+      !!(gtCount > 0) + !!(ltCount > 0),
+      1,
+      "gtCount {} ltCount {}",
+      gtCount,
+      ltCount);
+  return ltCount > 0;
```

Comment:
VELOX_CHECK(!(gtCount > 0 && ltCount > 0))?

Reply:
Both can be zero

Reply:
Yeah, simpler. Update.